.github/workflows/build_wheels.yml

```diff
@@ -8,31 +8,54 @@ on:
       - published
 
 env:
-  CIBW_ARCHS_MACOS: x86_64 arm64
+  CIBW_SKIP: cp36-* cp37-* cp38-* pp* *i686 *win32
+
+  # cibuildhweel macOS configuration
+  CIBW_ARCHS_MACOS: native
+
+  # cibuildhweel linux configuration
   CIBW_ARCHS_LINUX: x86_64
-  CIBW_BEFORE_ALL_LINUX: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain=stable --profile=minimal -y
+  CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_28
+  CIBW_MUSLLINUX_X86_64_IMAGE: musllinux_1_2
+  # On Linux, wheels are built inside a container on the GitHub runner (unlike macOS and Windows which use the GitHub runner directly)
+  # Any build dependencies must thus be installed on the container (using CIBW_BEFORE_ALL_LINUX), *not* on the runner with the regular workflow
+  # This command must be compatible with manylinux_2_28 (AlmaLinux 8) and musllinux_1_2 (Alpine Linux 3.20)
+  CIBW_BEFORE_ALL_LINUX: dnf install -y clang nasm; apk add clang-dev nasm; curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain=stable --profile=minimal -y
   CIBW_ENVIRONMENT_LINUX: 'PATH="$HOME/.cargo/bin:$PATH"'
-  CIBW_SKIP: cp36-* cp37-* cp38-* pp* *i686 *win32
+
+  # cibuildhweel Windows configuration
+  CIBW_ARCHS_WINDOWS: AMD64
 
 jobs:
   build_wheels:
     name: Build wheels on ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-    env:
-      wheelhouse_path: wheelhouse-${{ matrix.os }}
+        # macos-13 is an Intel runner, macos-latest is an ARM runner
+        os: [macos-13, macos-latest, ubuntu-latest, windows-latest]
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
-      - run: rustup target add aarch64-apple-darwin x86_64-apple-darwin
-        if: startsWith(matrix.os, 'macos')
-      - uses: pypa/cibuildwheel@v2.21.3
-      - run: echo "wheelhouse_path=wheelhouse-$(echo '${{ matrix.os }}' | awk '{split($0,input,"-"); print input[1]}')" >> $GITHUB_ENV
+      - run: python -m pip install cibuildwheel==2.21.3
+      - run: |
+          brew install nasm
```

Comment:
Does brew run on windows?

Reply:
Not to my knowledge. There's a handful of package managers on Windows. To cite a few:

- Chocolatey (installed by default on GitHub runners)
- MSYS2 (installed by default on GitHub runners but not in the path)
- Winget (official Microsoft package manager, not available by default on the current GitHub action runners)

I currently use Chocolatey to install NASM (the assembler that x264 uses during compilation). The other x264 compilation dependency (the Visual Studio compiler) is installed by default and just needs to be "sourced" to appear in the path (which is done by invoking the script `vcvars64.bat`).

Reply:
Urgh. I see. I'm not liking the complexity, but well. I guess that's what we get for now.